api/incidents/incidentsRouter.js

```diff
@@ -107,6 +107,40 @@ const { dsFetch } = require('../dsService/dsUtil');
  *                  type: string
  *                  example: "Request Error"
  */
+
+router.get('/queryincidents', async (req, res) => {
+  try {
+    const incidents = await Incidents.getAllIncidents();
+
+    const queryResponse = incidents.map((incident) => {
+      let dateObj = new Date(incident.date);
+      incident.src = JSON.parse(incident.src);
+      incident.categories = JSON.parse(incident.categories);
+      incident.date = dateObj.getMonth();
+      return incident;
+    });
+
+    let filterResponse = queryResponse.filter((item) => {
+      let searchStartDate = +req.query.startDate;
+      let searchEndDate = +req.query.endDate;
+
+      if (req.query.state) {
```

Comment:
Maybe add some comments here to clarify exactly what is being searched. It looks like it is only being searched by the month. And also what happens if the user selects the same month as the start and end date?

Reply:
I feel like the querying should be done in a model, not in the route. It should be neater and more performant.